pkg/scheduler/plugins/gpushare/utils.go

```diff
@@ -0,0 +1,224 @@
+		if gpuCore.Value() > 100 && gpuCore.Value()%100 != 0 {
+			return gpuCombination, fmt.Errorf("failed to validate koordinator.sh/gpu: %v", gpuCore)
+		}
+		gpuCombination |= gpuCoreExist
+	}
+	if _, exist := podRequest[apiext.GPUMemory]; exist {
+		gpuCombination |= gpuMemoryExist
+	}
+	if _, exist := podRequest[apiext.GPUMemoryRatio]; exist {
+		gpuCombination |= gpuMemoryRatioExist
+	}
+
+	if gpuCombination == (nvidiaGpuExist) ||
+		gpuCombination == (koordGpuExist) ||
+		gpuCombination == (gpuCoreExist|gpuMemoryExist) ||
+		gpuCombination == (gpuCoreExist|gpuMemoryRatioExist) {
+		return gpuCombination, nil
+	}
+
+	return gpuCombination, fmt.Errorf("request is not valid, current combination: %b", gpuCombination)
+}
+
+// nvidia.com/gpu means applying for full-card
+// koordinator.sh/gpu means applying for cards in percentile
+// convertGPUResource will convert either nvidia.com/gpu or koordinator.sh/gpu to koordinator.sh/gpu-core and koordinator.sh/gpu-memory-ratio
+func convertGPUResource(podRequest corev1.ResourceList, combination uint) corev1.ResourceList {
+	switch combination {
+	case gpuCoreExist | gpuMemoryExist:
+		return corev1.ResourceList{
+			apiext.GPUCore:   podRequest[apiext.GPUCore],
+			apiext.GPUMemory: podRequest[apiext.GPUMemory],
+		}
+	case gpuCoreExist | gpuMemoryRatioExist:
+		return corev1.ResourceList{
+			apiext.GPUCore:        podRequest[apiext.GPUCore],
+			apiext.GPUMemoryRatio: podRequest[apiext.GPUMemoryRatio],
+		}
+	case koordGpuExist:
+		return corev1.ResourceList{
+			apiext.GPUCore:        podRequest[apiext.KoordGPU],
+			apiext.GPUMemoryRatio: podRequest[apiext.KoordGPU],
+		}
+	case nvidiaGpuExist:
+		nvidiaGpu := podRequest[apiext.NvidiaGPU]
+		return corev1.ResourceList{
+			apiext.GPUCore:        *resource.NewQuantity(nvidiaGpu.Value()*100, resource.DecimalSI),
+			apiext.GPUMemoryRatio: *resource.NewQuantity(nvidiaGpu.Value()*100, resource.DecimalSI),
+		}
+	}
+	return nil
+}
+
+func checkNodeGPUSatisfied(podRequest corev1.ResourceList, nodeDeviceFree *deviceResources) bool {
+	if isMultipleGPUPod(podRequest) {
+		gpuCore, gpuMem, gpuMemRatio := podRequest[apiext.GPUCore], podRequest[apiext.GPUMemory], podRequest[apiext.GPUMemoryRatio]
+		gpuWanted := gpuCore.Value() % 100
+		podRequestPerCard := corev1.ResourceList{
+			apiext.GPUCore:        *resource.NewQuantity(gpuCore.Value()/gpuWanted, resource.DecimalSI),
+			apiext.GPUMemory:      *resource.NewQuantity(gpuMem.Value()/gpuWanted, resource.BinarySI),
+			apiext.GPUMemoryRatio: *resource.NewQuantity(gpuMemRatio.Value()/gpuWanted, resource.DecimalSI),
+		}
+
+		satisfiedDeviceCount := 0
+		for _, resources := range *nodeDeviceFree {
+			if satisfiedDeviceCount == int(gpuWanted) {
+				break
+			}
+			if satisfied, _ := quotav1.LessThanOrEqual(podRequestPerCard, resources); satisfied {
+				satisfiedDeviceCount++
+			}
+		}
+		return satisfiedDeviceCount == int(gpuWanted)
+	}
+
+	for _, resources := range *nodeDeviceFree {
```

Comment:
There is a special issue to consider. 

If NodeA has 7 GPU instances, the scheduler has allocated 6 instances to 6 Pods, and the corresponding GPU Minors are 0-4 and 6. At some point in the future, the remaining unallocated instance GPU Minor 5 were broken. At this time, only six GPU instances were found through the NVML, and the value of GPU Minor also changed. At this time, during Failover, restore the allocation result from the Annotation of the Pod, and you will find that GPU Minor 5 is not allocated in the view of the scheduler, but in fact the GPU Minor 5 at this time is the original Minor 6.

So we need to check whether the total amount of GPU remaining in the entire node dimension meets the requirements.

Reply:
Is there something like `device uid` to help us judge this situation?

Reply:
need some tests to verify

Reply:
gpu minor is a hardware property and won't change like uid.

Reply:
but that's exactly what happened.

Reply:
I'll mark this conversation as resolved temporarily. And figure out how to deal with it when user encounters this issue.